File: app/src/main/java/mhh/com/gotennaapp/repo/dao/PlacesDao.java
package mhh.com.gotennaapp.repo.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import java.util.List;
import mhh.com.gotennaapp.model.Place;

@Dao public interface PlacesDao {

    @Query("SELECT * FROM places") LiveData<List<Place>> getPlaces();

    @Insert(onConflict = OnConflictStrategy.REPLACE) void insertPlaces(List<Place> places);

    @Query("DELETE FROM places") void deleteAllPlaces();
}

File: app/src/main/java/mhh/com/gotennaapp/repo/dao/PlacesDao.kt
package mhh.com.gotennaapp.repo.dao

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import mhh.com.gotennaapp.model.Place

@Dao
interface PlacesDao {

    @get:Query("SELECT * FROM places")
    val places: LiveData<List<Place>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertPlaces(places: List<Place>)

    @Query("DELETE FROM places")
    fun deleteAllPlaces()
}
